openfisca_core/variables.py

```diff
@@ -219,9 +218,11 @@ def set(self, attributes, attribute_name, required = False, allowed_values = Non
         return value
 
     def set_entity(self, entity):
-        if not isinstance(entity, type) or not issubclass(entity, entities.Entity):
-            raise ValueError("Invalid value '{}' for attribute 'entity' in variable '{}'. Must be a subclass of Entity."
-            .format(entity, self.name).encode('utf-8'))
+        # TODO - isinstance() won't work due to (ab)use of load_module to load tax_benefit_system
```

Comment:
I'd rather avoid snarky comments about the codebase in the codebase. We can be factual about current limitations without explicit negative judgement 😉.

Reply:
I'm also able to uncomment the following lines without encountering any issue while loading country-template and france. In which case did you observe a problem?

Reply:
Fair enough, my bad! Removed the snark.

> In which case did you observe a problem?

Running the unit tests - the circumstance however is not necessarily limited to tests/CI. It's the same kind of problem we've encountered with Enums.

The case where a problem arises is when the class Entity is loaded independently, which currently happens at unit test time: two tests in `test_simulation_builder` fail because they attempt to set the `entity` of a variable to a subclass of Entity **imported directly by the test**, and because this ends up being a different module loader than the one for the country package, this subclass of Entity fails the `isinstance()` verification.

Reply:
Looking more closely, it seems that the issue was actually that we were still using an Entity class for a variable attribute in the unit test, rather than in instance. [This commit](https://github.com/openfisca/openfisca-core/pull/857/commits/cb75b93abc710dd841d4a2f04ac0d9859b2b1596) fixes it.

Reply:
For the record, I think the problem we've encountered with Enums is limited to classes _directly defined_ in the python module that is dynamically loaded, not imported.

Reply:
Good catch, thanks! The Entity class/instance situation had me confused.